### DISCUSSION TITLE
Name collision with Extension Methods in LinqExtensions Class

### DISCUSSION THREAD

**Discussion by akurone** [JasperFx/marten#2437] [General]
hi, 
first of all, thanks for this great work! a little shy about it but have a small suggestion (feel free to delete (if there is no such thing, ask me to delete) this issue completely): **can you consider taking `LinqExtensions` class a level down (e.g. like `Marten.Extensions`)?** this class has invaluable extensions for collections, in fact, many of these should already be in the BCL. my little problem is that we have already implemented some of them (like [`IsEmpty(this IEnumerable<T>? enumerable)`](https://github.com/JasperFx/marten/blob/5d32a4a12fbd787a96650e068418d8b0e46a90cc/src/Marten/LinqExtensions.cs#L125)) in our _toolbelt_ and adding `using Marten;` to the party (having 2 extension methods with same signature in two different classes in our _old_ and `Marten` namespace) upsets the compiler and it complains with [CS0121](https://learn.microsoft.com/en-us/dotnet/csharp/misc/cs0121). 

i know that CS0121 is easy to get around (and i have already done that) and this change will (most probably) be a breaking one but (what i think will add value to marten is:) making `Marten` namespace collusion free (as far as possible) would make the experience for potential users -especially from shops with common-sense tools like these already deployed- smoother. `global using` could make the impact smaller and having this extensions in a deeper namespace could give users more space for common names :)

kind regards,
ahmet.

**Comment by mysticmind** [comment]
We want to the extension methods to be easily discoverable and usable hence it is at the top level. I do understand your point about possible collisions but don't see a easier way out to resolve this.

@jeremydmiller  @oskardudycz Please chime in with your thoughts.

I am moving this to a discussion since it not a bug per se.
